pandas/core/strings.py

```diff
@@ -1953,8 +1953,11 @@ def _validate(data):
         values = getattr(data, "values", data)  # Series / Index
         values = getattr(values, "categories", values)  # categorical / normal
 
-        # missing values obfuscate type inference -> skip
-        inferred_dtype = lib.infer_dtype(values, skipna=True)
+        try:
+            inferred_dtype = lib.infer_dtype(values, skipna=True)
+        except ValueError:
```

Comment:
infer_dtype should *never* raise

Reply:
lib.pyx L1207 specifically raises ValueError

Reply:
ok, can you make an issue to fix the doc-string in infer_dtype. is possible to handle this *inside* infer_dtype rather than raise? or is this meant as a tactical fix?

Reply:
>  is possible to handle this inside infer_dtype rather than raise? or is this meant as a tactical fix?

the interval case is fixed by this PR inside infer_dtype, so catching the error here is just a dir-should-never-raise thing.  the more general infer_dtype fixes are part of a larger push to recognize EADtypes correctly, yah